File: src/main/java/com/oracle/ocmjea/gof/structural/adapter/AudioPlayer.java
package com.oracle.ocmjea.gof.structural.adapter;

public class AudioPlayer implements MediaPlayer {

	private MediaAdapter adapter;
	
	@Override
	public void play(String tipo, String arquivo) {
		switch (tipo) {
		case "mp3":
			System.out.println("Tocando arquivo MP3: " + arquivo);
			break;
		case "mp4":
			adapter = new MediaAdapter(tipo);
			adapter.play(tipo, arquivo);
			break;
		case "flv":
			adapter = new MediaAdapter(tipo);
			System.out.println("Tocando arquivo FLV: " + arquivo);
			break;
		default:
			System.out.println("Arquivo não suportado: " + arquivo);
			break;
		}
	}
}

File: design-patterns/src/main/kotlin/designpatterns/structural/adapter/AudioPlayer.kt
package designpatterns.structural.adapter

class AudioPlayer : MediaPlayer {
    private var adapter: MediaAdapter? = null

    override fun play(type: String, fileName: String) {
        when (type) {
            "mp3" -> println("Playing MP3 file: $fileName")
            "mp4" -> {
                adapter = MediaAdapter(type)
                adapter?.play(type, fileName)
            }
            "flv" -> {
                adapter = MediaAdapter(type)
                println("Playing FLV file: $fileName")
            }
            else -> println("Unsupported file: $fileName")
        }
    }
}
